codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/IdempotencyTokenMiddlewareGenerator.java

```diff
@@ -74,23 +105,47 @@ public void writeAdditionalFiles(
             SymbolProvider symbolProvider,
             TriConsumer<String, String, Consumer<GoWriter>> writerFactory
     ) {
-        ServiceShape serviceShape =  settings.getService(model);
+        ServiceShape serviceShape = settings.getService(model);
         Map<ShapeId, MemberShape> map = getOperationsWithIdempotencyToken(model, serviceShape);
         if (map.size() == 0) {
             return;
         }
 
-        writerFactory.accept("idempotencyTokenMiddleware.go", settings.getModuleName(), writer -> {
-            writer.write("var randReader io.Reader = cryptorand.Reader");
-            writer.addUseImports(SmithyGoDependency.IO);
-            for (Map.Entry<ShapeId, MemberShape> entry : map.entrySet()) {
-                ShapeId operationId = entry.getKey();
-                OperationShape operation = model.expectShape(operationId).asOperationShape().get();
-                Shape inputShape  = model.expectShape(operation.getInput().get());
-                MemberShape memberShape = entry.getValue();
-                execute(writer, symbolProvider, operation, memberShape, symbolProvider.toSymbol(inputShape));
-            }
+        writerFactory.accept("api_client.go", settings.getModuleName(), writer -> {
+            writer.write("// IdempotencyTokenProvider interface for providing idempotency token");
+            writer.openBlock("type IdempotencyTokenProvider interface {",
+                    "}", () -> {
+                        writer.write("GetToken() (string, error)");
+                    });
+            writer.write("");
```

Comment:
I think it might be better to use the GoDelegator directly. It should provide all of the capability write to specific files.

Reply:
Will have to introduce a new GoIntegration method but may not be needed as `writeAdditionalFiles` handles it.